src/modules/musig/tests_impl.h

```diff
@@ -539,23 +539,25 @@ void musig_api_tests(secp256k1_scratch_space *scratch) {
     CHECK(ecount == 3);
 
     ecount = 0;
-    {
-        unsigned char tmp_sig[64];
-        memcpy(tmp_sig, pre_sig, sizeof(tmp_sig));
-        CHECK(secp256k1_musig_adapt(none, tmp_sig, sec_adaptor, nonce_parity) == 1);
-        CHECK(secp256k1_musig_adapt(none, NULL, sec_adaptor, 0) == 0);
-        CHECK(ecount == 1);
-        CHECK(secp256k1_musig_adapt(none, max64, sec_adaptor, 0) == 0);
-        CHECK(ecount == 1);
-        CHECK(secp256k1_musig_adapt(none, tmp_sig, NULL, 0) == 0);
-        CHECK(ecount == 2);
-        CHECK(secp256k1_musig_adapt(none, tmp_sig, max64, nonce_parity) == 0);
-        CHECK(ecount == 2);
-        CHECK(secp256k1_musig_adapt(none, tmp_sig, sec_adaptor, 2) == 0);
-        CHECK(ecount == 3);
-    }
+    CHECK(secp256k1_musig_adapt(none, final_sig, pre_sig, sec_adaptor, nonce_parity) == 1);
+    CHECK(secp256k1_musig_adapt(none, NULL, pre_sig, sec_adaptor, 0) == 0);
+    CHECK(ecount == 1);
+    CHECK(secp256k1_musig_adapt(none, final_sig, NULL, sec_adaptor, 0) == 0);
+    CHECK(ecount == 2);
+    CHECK(secp256k1_musig_adapt(none, final_sig, max64, sec_adaptor, 0) == 0);
+    CHECK(ecount == 2);
+    CHECK(secp256k1_musig_adapt(none, final_sig, pre_sig, NULL, 0) == 0);
+    CHECK(ecount == 3);
+    CHECK(secp256k1_musig_adapt(none, final_sig, pre_sig, max64, 0) == 0);
+    CHECK(ecount == 3);
+    CHECK(secp256k1_musig_adapt(none, final_sig, pre_sig, sec_adaptor, 2) == 0);
+    CHECK(ecount == 4);
+    /* sig and pre_sig argument point to the same location */
```

Comment:
So much for `const` 🤣

Reply:
Well yeah, `const` in a function sig is just a promise not to write through that pointer. And we keep that promise.   

On a more serious note, I think now we should document in the API docs that `sig` and `pre_sig` may overlap.

Reply:
> On a more serious note, I think now we should document in the API docs that `sig` and `pre_sig` may overlap/alias.

This part is still unresolved (if you agree).

Reply:
Added your suggested sentence